src/subscription/subscription.controller.ts

```diff
@@ -36,26 +36,30 @@ export class SubscriptionController {
         );
     }
 
-    @Get('users/:plainTextEmailAddress/grants/:grantId')
-    async findByEmailAddressAndGrantId(
-        @Param('plainTextEmailAddress') plainTextEmailAddress: string,
+    @Get('users/:id/grants/:grantId')
+    async findByUserAndGrantId(
+        @Param('id') id: string,
         @Param('grantId') grantId: string,
     ): Promise<Subscription> {
-        return this.subscriptionService.findByEmailAndGrantId(
-            plainTextEmailAddress,
+        const subscription = await this.subscriptionService.findBySubAndGrantId(
+            id,
             grantId,
         );
+
+        return subscription
+            ? subscription
+            : this.subscriptionService.findByEmailAndGrantId(id, grantId);
     }
 
-    @Delete('users/:plainTextEmailAddress/grants/:grantId')
-    async deleteByEmailAndGrantId(
-        @Param('plainTextEmailAddress') plainTextEmailAddress: string,
+    @Delete('users/:id/grants/:grantId')
```

Comment:
This endpoint is currently consumed by the front-end unsub page `gap-find-web/pages/unsubscribe/[id].tsx` Needs updated there too